flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/batch/sql/FunctionITCase.java

```diff
@@ -0,0 +1,167 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.flink.table.planner.runtime.batch.sql;
+
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.planner.factories.utils.TestCollectionTableFactory;
+import org.apache.flink.table.planner.runtime.utils.BatchTestBase;
+import org.apache.flink.table.resource.ResourceManagerTest;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.UserClassLoaderJarTestUtils;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** Tests for catalog and system functions in a table environment. */
+public class FunctionITCase extends BatchTestBase {
+
+    private static String jarPath;
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        File jarFile =
+                UserClassLoaderJarTestUtils.createJarFile(
+                        TEMPORARY_FOLDER.newFolder("test-jar"),
+                        "test-classloader-udf.jar",
+                        ResourceManagerTest.LOWER_UDF_CLASS,
+                        ResourceManagerTest.LOWER_UDF_CODE);
+        jarPath = jarFile.toURI().toString();
+    }
+
+    @Test
+    public void testCreateTemporarySystemFunctionByUsingJar() {
+        String ddl =
+                String.format(
+                        "CREATE TEMPORARY SYSTEM FUNCTION f10 AS '%s' USING JAR '%s'",
+                        ResourceManagerTest.LOWER_UDF_CLASS, jarPath);
+        tEnv().executeSql(ddl);
```

Comment:
It seems you use the same ResourceManager in all tests. If so, the created jar will only be added for the first test?

Reply:
good catch